test/module/shared_model/backend_proto/proto_query_response_factory_test.cpp

```diff
@@ -0,0 +1,294 @@
+#include "backend/protobuf/proto_query_response_factory.hpp"
+#include "cryptography/crypto_provider/crypto_defaults.hpp"
+#include "framework/specified_visitor.hpp"
+#include "module/shared_model/builders/protobuf/test_block_builder.hpp"
+#include "module/shared_model/builders/protobuf/test_transaction_builder.hpp"
+#include "validators/field_validator.hpp"
+
+using namespace shared_model::proto;
+using namespace iroha::expected;
+using namespace shared_model::interface::types;
+
+using framework::SpecifiedVisitor;
+using shared_model::validation::FieldValidator;
+
+class ProtoQueryResponseFactoryTest : public ::testing::Test {
+ public:
+  std::shared_ptr<ProtoQueryResponseFactory> response_factory =
+      std::make_shared<ProtoQueryResponseFactory>();
+  std::shared_ptr<ProtoCommonObjectsFactory<FieldValidator>> objects_factory =
+      std::make_shared<ProtoCommonObjectsFactory<FieldValidator>>();
+
+  /**
+   * Put value of Result<unique_ptr<_>, _> into a shared_ptr
+   * @tparam ResultType - type of result value inside a unique_ptr
+   * @tparam ErrorType - type of result error
+   * @param res - result to be unwrapped
+   * @return shared_ptr to result value or to nullptr, if result containts error
+   */
+  template <typename ResultType, typename ErrorType>
+  std::shared_ptr<ResultType> unwrapResult(
+      Result<std::unique_ptr<ResultType>, ErrorType> &&res) {
+    return res.match(
+        [](Value<std::unique_ptr<ResultType>> &val) {
+          std::shared_ptr<ResultType> ptr = std::move(val.value);
+          return ptr;
+        },
+        [](const Error<ErrorType> &) -> std::shared_ptr<ResultType> {
+          return nullptr;
+        });
+  }
+
+  void SetUp() override {}
+  void TearDown() override {}
+};
+
+/**
+ * Checks createAccountAssetResponse method of QueryResponseFactory
+ * @given collection of account assets
+ * @when creating account asset query response via factory
+ * @then that response is created @and is well-formed
+ */
+TEST_F(ProtoQueryResponseFactoryTest, CreateAccountAssetResponse) {
+  constexpr int kAccountAssetsNumber = 5;
+  const std::string kAccountId = "doge@meme";
+  const std::string kAssetId = "dogecoin#iroha";
+
+  std::vector<std::shared_ptr<shared_model::interface::AccountAsset>> assets;
+  for (auto i = 1; i < kAccountAssetsNumber; ++i) {
+    auto asset = unwrapResult(objects_factory->createAccountAsset(
+        kAccountId,
+        kAssetId,
+        shared_model::interface::Amount(std::to_string(i))));
+    if (not asset) {
+      FAIL() << "could not create common object via factory";
+    }
+    assets.push_back(asset);
+  }
+  auto response = response_factory->createAccountAssetResponse(assets);
+
+  ASSERT_TRUE(response);
+  ASSERT_EQ(response->accountAssets().front().accountId(), kAccountId);
+  ASSERT_EQ(response->accountAssets().front().assetId(), kAssetId);
+  for (auto i = 1; i < kAccountAssetsNumber; i++) {
+    ASSERT_EQ(response->accountAssets()[i - 1].balance(),
```

Comment:
Maybe just start with 0?

Reply:
I can, but in that case this 1 will be always added to balance, because the former cannot be equal to zero. I think, there's no difference